docs/ape17.rst

```diff
@@ -0,0 +1,533 @@
+and implications of this are discussed in an Astropy Proposal for
+Enhancements: `APE 17: A roadmap for package infrastructure without
+astropy-helpers <https://github.com/astropy/astropy-APEs/blob/master/APE17.rst>`__
+
+This page aims to provide a guide to migrating to using the new infrastructure
+described in APE 17. We assume that your package is currently using
+astropy-helpers and that you have used a version of the Astropy package-template
+in the past to set up your package (though this guide might still be useful if
+the latter is not the case).
+
+Throughout this guide, we will assume that your package is called ``my-package``
+and that the module is called ``my_package``. We deliberately choose a name
+where the package name is different from the module name, but for many cases,
+these will be the same.
+
+.. _step0:
+
+Step 0: Re-rendering the template
+---------------------------------
+
+In this guide, we will descibe the changes to make to the files in your package.
+To make some of the steps easier below, you should first re-run cookiecutter to
+re-render the template to a temporary folder::
+
+    cookiecutter gh:astropy/package-template -o my_package_tmp
+
+We will refer to some of the rendered files in several of the steps below.
+
+Step 1: Remove astropy-helpers
+------------------------------
+
+To remove the astropy-helpers submodule, first, run::
+
+   git rm -r astropy_helpers
+
+if ``astropy_helpers`` was the only submodule in your repository, the
+``.gitmodules`` file will be empty, you can remove this if you wish.
+
+Next you should remove the ``ah_bootstrap.py`` file::
+
+   git rm ah_bootstrap.py
+
+You can now commit your changes with::
+
+   git commit -m "Remove astropy-helpers submodule"
+
+Step 2: Update/create ``setup.cfg``
+-----------------------------------
+
+The next step is to update make sure that you have a ``setup.cfg`` file
+that contains meta-data about the package. If you already have this
+file, you will likely need to update it, and if you don’t already have
+this file, you will need to create it.
+
+This file should contain at least the following entries::
+
+   [metadata]
+   name = my-package
+   author = ...
+   author_email = ...
+   license = ...
+   license_file = LICENSE.rst
+   url = ...
+   description = ...
+   long_description = file: README.rst
+
+   [options]
+   zip_safe = False
+   packages = find:
+   install_requires =
+       numpy
+       astropy
+       ...
+   python_requires = >=3.6
+
+Replace the ``...`` with the information for your package. Make sure
+that ``license_file`` and ``long_description`` have the right filename,
+and specify your required dependencies one per line in the
+``install_requires`` section. Make sure the ``python_requires`` line is
+set to indicate the correct minimum Python version for your package.
+
+If you already had a file, make sure you remove the following entries
+(if present):
+
+-  ``package_name``
+-  ``version``
+-  ``setup_requires``
```

Comment:
Removing `setup_requires` directly contradicts what is in Step 8.

Reply:
I've remove it from the list, but I've added back a mention to remove any existing ``setup_requires`` entry and that we'll add it back in Step 8. The point here is that any existing setup_requires with e.g. numpy or cython is no longer necessary.